src/wildcard.cpp

```diff
@@ -374,7 +434,7 @@ static const wchar_t *file_get_desc(const wcstring &filename, int lstat_res,
         return COMPLETE_SOCKET_DESC;
     } else if (S_ISDIR(buf.st_mode)) {
         return COMPLETE_DIRECTORY_DESC;
-    } else if (buf.st_mode & (S_IXUSR | S_IXGRP | S_IXGRP) && waccess(filename, X_OK) == 0) {
```

Comment:
I'm pretty sure this is a bug and was supposed to be `S_IXUSR | S_IXGRP | S_IXOTH`